src/CoreShop/Component/Core/Order/Transformer/CartItemToSaleItemTransformer.php

```diff
@@ -30,10 +24,28 @@ public function __construct(ProposalItemTransformerInterface $innerCartItemToSal
         $this->innerCartItemToSaleItemTransformer = $innerCartItemToSaleItemTransformer;
     }
 
+    /**
+     * @param ProposalInterface     $proposal
+     * @param ProposalItemInterface $fromProposalItem
+     * @param ProposalItemInterface $toProposal
+     * @return mixed
+     */
     public function transform(ProposalInterface $proposal, ProposalItemInterface $fromProposalItem, ProposalItemInterface $toProposal)
     {
-        if ($fromProposalItem instanceof ProductInterface && $toProposal instanceof ProductInterface) {
+        if ($fromProposalItem instanceof CartItemInterface && $toProposal instanceof SaleItemInterface) {
             $toProposal->setDigitalProduct($fromProposalItem->getDigitalProduct());
+            $toProposal->setObjectId($fromProposalItem->getProduct()->getId());
+            $mainObjectId = null;
+
+            if ($fromProposalItem->getProduct() instanceof ProductInterface) {
+                if ($fromProposalItem->getProduct()->getType() === 'variant') {
+                    $mainProduct = $fromProposalItem->getProduct();
+                    while ($mainProduct->getType() === 'variant') {
```

Comment:
wouldn't it make sense to add a getContainerProduct function inside the Product model? I guess we'll need that anyway somewhere else.

Reply:
mhm... there is already a method called getVariantMaster()

Reply:
does it what its supposed to do?

Reply:
yes :).